armi/reactor/reactors.py

```diff
@@ -2129,6 +2129,48 @@ def growToFullCore(self, cs):
 
         return converter
 
+    def calculateNewGridPlatePitchFromTemp(self, cs, isoTemp: float) -> float:
+        """calculate a grid plate pitch using a new bulk ave core inlet temperature in C"""
+        # retrieve the first grid plate (all the same material, so just get the first)
+        gridPlateBlock = self.getFirstBlock(blockType=Flags.GRID_PLATE)
```

Comment:
Since there is nothing currently enforcing that all GRID_PLATE blocks have the same material, it might be worthwhile to include a check here and throw a warning, if not.

Reply:
https://github.com/terrapower/armi/pull/993/commits/191f193736fd82a9ad91ae6e3d7080156c86dd08